depthmapXcli/runmethods.cpp

```diff
@@ -53,6 +54,13 @@ namespace dm_runmethods
         return mgraph;
     }
 
+    std::unique_ptr<Communicator> getCommunicator(const CommandLineParser &clp) {
+        if (clp.printProgress()) {
+            return std::unique_ptr<Communicator>(new PrintCommunicator());
```

Comment:
I am not sure if the unique pointer is required here - wouldn't returning a raw pointer wrap it automatically? But probably doesn't make a difference as the return value optimisation will ensure there's just one unique pointer created and moved.